### ISSUE TITLE
Have workspace Resolution functionality along with Node Module Resolution for dependency resolution

### ISSUE THREAD

**Issue by knarwani** [microsoft/rnx-kit#2457]
**What I need?**
Can we please have the workspace resolution functionality as well along with node_module resolution of the dependencies? 

**Why I need?**
I am using rnx-kit for aligning dependencies in my project. I am working on a monorepo and I wanted to use align-deps before installing any dependencies in my project (To run rnx-kit I have created a bundle of it and calling it). For that I need workspace resolution of few of my internal packages which are yet not available in node_modules. So, can we please have the workspace resolution functionality as well along with node_module resolution of the dependencies?

**Where to change?**
I think [this](https://github.com/microsoft/rnx-kit/blob/main/packages/align-deps/src/preset.ts#L24) is the place where my functionality can be added.

**Comment by krishna8770** [comment]
Hi there, 

I wanted to change [this](https://github.com/microsoft/rnx-kit/blob/main/packages/align-deps/src/preset.ts) file where loadPreset function will change to 
```
function loadPreset(
  preset: string,
  projectRoot: string,
  resolve = require.resolve
): Preset {
  switch (preset) {
    case "microsoft/react-native":
      return reactNativePreset;
    default: {
      const info = loadInfo(preset);
      if (info !== undefined) {
        const returnResolvedPath = path.join(path.dirname(info.packageJsonPath), parsedInfo.path);
        return require(returnResolvedPath);
      }
      else {
        return require(resolve(preset, { paths: [projectRoot] }));
      }
    }
  }
}
```

and will add this const in [config.ts](https://github.com/microsoft/rnx-kit/blob/main/packages/align-deps/src/config.ts)

```
export const allPackages = getPackageInfos(process.cwd());
```

I tried this solution in my local and is working fine. Can you have a look at this code and let me know if I can go ahead and create a PR for this change.

Please let me know what your views on this are.

Thanks.

**Comment by krishna8770** [comment]
Yes, the expensiveness of the function was my concern also but, to address that I have exported this variable from [config.ts](https://github.com/microsoft/rnx-kit/blob/main/packages/align-deps/src/config.ts) file i.e., ``` export const allPackages = getPackageInfos(process.cwd()); ``` which then will be calculated only once and then passed on to further iterations. This solution is not at all increasing the runtime on my monorepo with approx 400 packages. Let me know if you are fine with this solution being merged.

**Comment by tido64** [comment]
We should not use global states as we have no control over how this package may be used. Our best bet is to pipe through the `getManifests` result to `loadPreset` as I've mentioned in https://github.com/microsoft/rnx-kit/issues/2457#issuecomment-1642154422.

**Comment by krishna8770** [comment]
I have made the subsequent changes. Can you please let me know how to create PR on this repo as I am not getting option to create new branch with either of my accounts (Microsoft or personal).

**Comment by tido64** [comment]
> I have made the subsequent changes. Can you please let me know how to create PR on this repo as I am not getting option to create new branch with either of my accounts (Microsoft or personal).

Please follow this guide: https://docs.github.com/en/get-started/quickstart/contributing-to-projects

**Comment by krishna8770** [comment]
[Here](https://github.com/microsoft/rnx-kit/pull/2550) is the PR for the change. I had to make a lot of changes even in the test files. Can you please review this once (There are few pipeline check issues, please ignore them for now I am working on them)?

**Comment by tido64** [comment]
Closing since there's no demand and the PR has gone stale.
